worker/transfer.go

```diff
@@ -338,16 +340,21 @@ func parallelDownloadSlab(ctx context.Context, hp hostProvider, ss object.SlabSl
 			contract := contracts[hostIndex]
 			shard := &ss.Shards[r.shardIndex]
 
-			err := hp.withHostV3(ctx, contract.ID, contract.HostKey, contract.SiamuxAddr, func(ss hostV3) error {
-				buf := bytes.NewBuffer(make([]byte, 0, rhpv2.SectorSize))
-				err := ss.DownloadSector(ctx, buf, shard.Root, r.offset, r.length)
+			buf := bytes.NewBuffer(make([]byte, 0, rhpv2.SectorSize))
+			err := func() error {
```

Comment:
this inline function is no longer necessary or useful, would remove it

Reply:
I added that to avoid sending to the channel twice since I thought I remembered you saying you don't want to send to the channel multiple times to avoid errors.